Added fades for 3P4P menuing

Varuuna opened the pull request:
Fixes #526 

The Lakitu screen fades are also used in the menu. This meant that when 3P or 4P menu options were selected, the related cases would be hit, which has been modified compared to decomp (I assume to better handle transitions when racing in wider formats).

Adding another `else if` which checks for `arg0 == 4` (which seems to be the menu) and drawing our original `draw_box_wide` seems to fix this issue.

Before:
![before](https://github.com/user-attachments/assets/be694b43-7146-4f50-a769-c34734e7374f)

After:
![after](https://github.com/user-attachments/assets/5290eb98-5d90-4e90-833a-2a07fd01c6fa)

coco875 commented:
did you compare to the decomp code ?

coco875 reviewed (commented):
hmm if it work then I guess it's fine ? but maybe add a comment to say that it's a bit weird in the logical sense ?

Varuuna commented:
> hmm if it work then I guess it's fine ? but maybe add a comment to say that it's a bit weird in the logical sense ?

I added a comment on the fade-in/out about it.
Reverted the change on the "single frame full black" fade because that only happens during Lakitu tow and not during the menu transitions.
